docs/tasks/argo.rst

```diff
@@ -0,0 +1,253 @@
+
+.. code-block:: bash
+
+  kanctl create actionset --action backup --namespace kanister --blueprint [YOUR_BLUEPRINT_NAME] --statefulset [YOUR_STATEFULSET] --profile [YOUR_PROFILE_NAME] --secrets [YOUR_SECRETS_NAME]
+
+Then execute -
+
+.. code-block:: bash
+
+  cat<<EOF >> mysql-cron-wf.yaml
+  apiVersion: argoproj.io/v1alpha1
+  kind: CronWorkflow
+  metadata:
+    name: mysql-cron-wf
+  spec:
+    schedule: "*/5 * * * *"
+    concurrencyPolicy: "Replace"
+    workflowSpec:
+      entrypoint: automate-actionset
+      templates:
+        - name: automate-actionset
+          container:
+            image: ghcr.io/kanisterio/kanister-tools:v9.99.9-dev
+            command:
+              - /bin/bash
+              - -c
+              - |
+                microdnf install tar
+                curl -LO https://github.com/kanisterio/kanister/releases/download/0.80.0/kanister_0.80.0_linux_amd64.tar.gz
+                tar -C /usr/local/bin -xvf kanister_0.80.0_linux_amd64.tar.gz
+                kanctl create actionset --action backup --namespace kanister --blueprint mysql-blueprint --statefulset mysql-test/mysql-release --profile mysql-test/s3-profile-gd4kx --secrets mysql=mysql-test/mysql-release
+  EOF
+
+Note - Here, the cron-job is scheduled to run every 5 minutes. You may schedule
+it to run as per your requirements.
+
+Step 4 - Granting RBAC permissions.
+````````````````````````````````````````````
+
+Next, you will grant the Service Account in our ``argo`` namespace to access resources
+in the ``kanister`` and ``mysql-test`` namespace. This is required to create CRs based on
+the Secrets and StatefulSets that you provided in our previous step.
+You may read more about RBAC authorization here - RBAC_.
+
+1. Create a ClusterRole named ``cron-wf-manager`` having the permissions in
+   ``cluster-admin``.
+2. Grant these permissions to the default ServiceAccount named
+   ``default`` in the ``argo`` namespace.
+3. This will be done by creating a RoleBinding in the ``kanister``
+   and ``mysql-test`` namespaces.
+
+Execute the following -
+
+.. code-block:: bash
+
+  kubectl create rolebinding cron-wf-manager --clusterrole=cluster-admin --serviceaccount=argo:default -n kanister
+
+.. code-block:: bash
+
+  kubectl create rolebinding cron-wf-manager --clusterrole=cluster-admin --serviceaccount=argo:default -n mysql-test
+
+Step 5 - Launching the Cron Workflow
+````````````````````````````````````````````
+
+Lets launch the workflow in the ``argo`` namespace by running -
+
+.. code-block:: bash
+
+  argo cron create mysql-cron-wf.yaml -n argo
+
+Check if the workflow was created by running -
+
+.. code-block:: bash
+
+  argo cron list -n argo
+
+When the workflow runs, check if the ActionSet was created in the ``kanister`` namespace -
+
+.. code-block:: bash
+
+  kubectl get actionsets.cr.kanister.io -n kanister
+
+.. image:: img/argo-cron-created.png
+
+Here, the workflow was created and scheduled to run in 1 minute. After it ran
```

Comment:
At some place I think we said that we are going to run the workflow every 5 minutes. Here we are saying 1 minutes. is it correct?
Also, we should specify what is the difference between `argo cron create` and `argo cron submit`.

Reply:
1. It will run every 5 minutes but the 1st one is scheduled to run under 1 minute.
2. Their documentation doesn't mention the difference but I believe that creation simply creates the workflow object and submit puts it into action. Now there are 2 things to consider here -
      a. This is a Cron workflow which is why it began execution immediately (according to my understanding).
      b. If a workflow consisted of other workflows, so sort of like a nested set of steps, submitting is required for all of 
          them.
Considering these points I did not include the difference between create in submit.

Reply:
Done.

Reply:
1.  :+1:  Ok, can they check somehow, where this 1 minute is mentioned/configured.
2. If by just creating the workflow we are making sure that actionset are going to get created after every 5 minutes. Is there a need to submit the workflow? If not, can we just mention that there is something else that is submitting the workflow and you can read more about that following the docs.

Reply:
1. Done.
2. Discussed the ambiguity in Argo docs for submit and create with Vivek.